runtimes/alibi-detect/mlserver_alibi_detect/protocols/v2.py

```diff
@@ -0,0 +1,62 @@
+from .request_handler import RequestHandler
+from mlserver.errors import InferenceError
+import numpy as np
+
+
+def _create_np_from_v2(data: list, ty: str, shape: list) -> np.array:
+    npty = np.float
+    if ty == "BOOL":
+        npty = np.bool
+    elif ty == "UINT8":
+        npty = np.uint8
+    elif ty == "UINT16":
+        npty = np.uint16
+    elif ty == "UINT32":
+        npty = np.uint32
+    elif ty == "UINT64":
+        npty = np.uint64
+    elif ty == "INT8":
+        npty = np.int8
+    elif ty == "INT16":
+        npty = np.int16
+    elif ty == "INT32":
+        npty = np.int32
+    elif ty == "INT64":
+        npty = np.int64
+    elif ty == "FP16":
+        npty = np.float32
+    elif ty == "FP32":
+        npty = np.float32
+    elif ty == "FP64":
+        npty = np.float64
+    else:
+        raise InferenceError(f"V2 unknown type or type that can't be coerced {ty}")
+
+    arr = np.array(data, dtype=npty)
+    arr.shape = tuple(shape)
+    return arr
+
+
+class KFservingV2RequestHandler(RequestHandler):
```

Comment:
For this one, we could re-use the existing `NumpyCodec` that we have in the `codecs.numpy` module:

https://github.com/SeldonIO/MLServer/blob/f088620d67cd366ad19857db38d5f89c7586a427/mlserver/codecs/numpy.py#L42-L64

It should be just a question of calling `NumpyCodec().decode(request.inputs[0])`, and that should take care of unwrapping the request into an array.